include/mbedtls/mbedtls_config.h

```diff
@@ -3916,4 +3916,9 @@
  */
 //#define MBEDTLS_ECDH_VARIANT_EVEREST_ENABLED
 
+/**
+ * Platform independent implementation for crypto algorithms.
```

Comment:
That does not match the logic of this PR.

- MBEDTLS_AESCE_C, enable AESCE
- MBEDTLS_AESNI_C, enable AESNI
- MBEDTLS_HAS_NO_BUILTIN,disable pure-c code

> * If MBEDTLS_AESCE_C, MBEDTLS_AESNI_C or MBEDTLS_PADLOCK_C is set, the plain C implementation will be removed, saving code size.

`the plain C implementation` will be removed if `MBEDTLS_HAS_NO_BUILTIN` is set. Otherwise it will not be removed.

> * Code that uses CPU acceleration will always be used, regardless of whether the CPU actually supports the feature.

In this PR, I do not think we should always uses CPU acceleration. I think the security issue should be a error in `check_config.h`

Reply:
This PR followup some ideas from #7004.

Reply:
Updated, but not follow your suggestion.